x/gov/client/rest/rest.go

```diff
@@ -0,0 +1,202 @@
+	"github.com/cosmos/cosmos-sdk/wire"
+	"github.com/cosmos/cosmos-sdk/x/gov"
+	"github.com/gorilla/mux"
+	"github.com/pkg/errors"
+	"github.com/tendermint/go-crypto/keys"
+)
+
+// RegisterRoutes - Central function to define routes that get registered by the main application
+func RegisterRoutes(ctx context.CoreContext, r *mux.Router, cdc *wire.Codec, kb keys.Keybase) {
+	r.HandleFunc("/gov/proposal", postProposalHandlerFn(cdc, kb, ctx)).Methods("POST")
+	r.HandleFunc("/gov/deposit", depositHandlerFn(cdc, kb, ctx)).Methods("POST")
+	r.HandleFunc("/gov/vote", voteHandlerFn(cdc, kb, ctx)).Methods("POST")
+	r.HandleFunc("/gov/{proposalID}/proposal", queryProposalHandlerFn("gov", cdc, kb, ctx)).Methods("GET")
+	r.HandleFunc("/gov/{proposalID}/votes/{voterAddress}", queryVoteHandlerFn("gov", cdc, kb, ctx)).Methods("GET")
+}
+
+func postProposalHandlerFn(cdc *wire.Codec, kb keys.Keybase, ctx context.CoreContext) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		var req postProposalReq
+		err := buildReq(w, r, &req)
+		if err != nil {
+			return
+		}
+
+		if !req.Validate(w) {
+			return
+		}
+
+		proposer, err := sdk.GetAccAddressBech32(req.Proposer)
+		if err != nil {
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte(err.Error()))
+			return
+		}
+
+		// create the message
+		msg := gov.NewMsgSubmitProposal(req.Title, req.Description, req.ProposalType, proposer, req.InitialDeposit)
+
+		// sign
+		signAndBuild(w, ctx, req.BaseReq, msg, cdc)
+	}
+}
+
+func depositHandlerFn(cdc *wire.Codec, kb keys.Keybase, ctx context.CoreContext) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		var req depositReq
+		err := buildReq(w, r, &req)
+		if err != nil {
+			return
+		}
+
+		if !req.Validate(w) {
+			return
+		}
+
+		depositer, err := sdk.GetAccAddressBech32(req.Depositer)
+		if err != nil {
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte(err.Error()))
+			return
+		}
+
+		// create the message
+		msg := gov.NewMsgDeposit(depositer, req.ProposalID, req.Amount)
+
+		// sign
+		signAndBuild(w, ctx, req.BaseReq, msg, cdc)
+	}
+}
+
+func voteHandlerFn(cdc *wire.Codec, kb keys.Keybase, ctx context.CoreContext) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		var req voteReq
+		err := buildReq(w, r, &req)
+		if err != nil {
+			return
+		}
+
+		if !req.Validate(w) {
+			return
+		}
+
+		voter, err := sdk.GetAccAddressBech32(req.Voter)
+		if err != nil {
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte(err.Error()))
+			return
+		}
+
+		// create the message
+		msg := gov.NewMsgVote(voter, req.ProposalID, req.Option)
+		// sign
+		signAndBuild(w, ctx, req.BaseReq, msg, cdc)
+	}
+}
+func queryProposalHandlerFn(storeName string, cdc *wire.Codec, kb keys.Keybase, ctx context.CoreContext) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		vars := mux.Vars(r)
+		strProposalID := vars["proposalID"]
```

Comment:
where are the `vars` inputs being defined? kind of confused by how this is being used

Reply:
Can you explain what you mean?

Reply:
like the words "proposalID" should be held in a string object somewhere and referenced throughout instead of manually typing "proposalID" (which is subject to typos)